ztrouy/gondor-backend
# Feature - Fetch Recent Patient Data

### Purpose
To allow Clinicians to get the most recent vitals of their Patients.

### Files Changed
- Renamed the views `user.py`, `appointment.py`, `state.py`, and `address.py` to be plural
- Added method that allows a provider to get the most recent Patient vitals in `users.py`
- Fixed incorrect first and last name properties for Tyler Hillard in `users.json`

### To Test
Fetch, then run the following commands in the project root directory
```
pipenv shell
```
```
pipenv install
```
```
./seed_database.sh
```
Then confirm the following:
- As a non-Clinician, a GET request to `/users/{userId}/records/last` returns 403 FORBIDDEN
- As a Clinician, a GET request to `/users/{userId}/records/last` for a Patient that they do not have a Patient/Clinician relationship with returns 403 FORBIDDEN
- As a Clinician, a GET request to `/users/{userId}/records/last` with an Id of a non-Patient returns 400 BAD REQUEST
- As a Clinician, a GET request to `/users/{userId}/records/last` to a Patient without any Patient Data returns 404 NOT FOUND
- As a Clinician, a GET request to `/users/{userId}/records/last` to a Patient you have a Patient/Clinician relationship with returns the weight, systolic, and diastolic recorded of their most recent Patient Data
